### PR TITLE
Halves the chance of floor tile ripping.

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Chance of decompression ripping floor tiles is halved, as is the maximum chance of ripping floor tiles.
Max chance is now 15% per tile, down from 30%.

## Why It's Good For The Game

Decompression still rips floor tiles, it's less tedious to fix afterwards.

## Changelog
:cl:
tweak: Floor tile ripping due to decompression now half as likely.
/:cl:

### CODE REVIEW

**Comment by Vexylius** [comment]
HELL YEAH! 😎

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 1]
```diff
@@ -185,7 +185,7 @@
 
 /turf/proc/handle_decompression_floor_rip()
 /turf/open/floor/handle_decompression_floor_rip(sum)
```
Comment (Jalleo) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]: Uh I think this proc should not be calling remove_tile. There is actually another proc it should be calling. remove_tile is meant for players not atmos. Is this something that monster coded? Also line above this is likely not actually needed.

Comment (Jalleo) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]: Actually line above this should have a return. I am going to investigate the blame for the potential of it being related to a exploit that I am aware of at the moment.

Comment (Jalleo) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]:
```suggestion
/turf/open/floor/handle_decompression_floor_rip(sum)
	if(sum > 20 && prob(CLAMP(sum / 20, 0, 15)))
		new floor_tile(src)
		make_plating()

/turf/open/floor/engine/handle_decompression_floor_rip()
	return
```

Comment (Jalleo) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]: The above fixes a specific issue and uses the proper method that most floor interactions have for making tiles. This is a error made by monster which opened up certain unintended interactions.

Comment (ike709) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]: @Jalleo fix the values to not undo the PR's changes and I'll commit your suggestion

Comment (Jalleo) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]: Done. It was a mistake my end I thought I fixed it already before mentioning about it.

Comment (Jalleo) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm:187]: Noticed a issue fixing quickly then its resolved.

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 2]
```diff
@@ -185,7 +185,13 @@
...
-	if(sum > 20 && prob(CLAMP(sum / 10, 0, 30)))
+	if(sum > 20 && prob(CLAMP(sum / 20, 0, 15)))
+		new floor_tile(src)
+		make_plating()
+
+/turf/open/floor/engine/handle_decompression_floor_rip()
+	return
+	if(sum > 20 && prob(CLAMP(sum / 20, 0, 15)))
 		remove_tile()
```
Comment (ike709) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm]:
```suggestion
```

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 3]
```diff
@@ -185,8 +185,12 @@
 
 /turf/proc/handle_decompression_floor_rip()
 /turf/open/floor/handle_decompression_floor_rip(sum)
-	if(sum > 20 && prob(CLAMP(sum / 10, 0, 30)))
-		remove_tile()
+	if(sum > 20 && prob(CLAMP(sum / 20, 0, 15)))
+		new floor_tile(src)
+		make_plating()
+
+/turf/open/floor/engine/handle_decompression_floor_rip()
```
Comment (Jalleo) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:196]:
```suggestion
/turf/open/floor/plating/handle_decompression_floor_rip()
	return

/turf/open/floor/engine/handle_decompression_floor_rip()
```

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 4]
```diff
@@ -185,8 +185,15 @@
 
 /turf/proc/handle_decompression_floor_rip()
 /turf/open/floor/handle_decompression_floor_rip(sum)
-	if(sum > 20 && prob(CLAMP(sum / 10, 0, 30)))
-		remove_tile()
+	if(sum > 20 && prob(CLAMP(sum / 20, 0, 15)))
+		new floor_tile(src)
```
Comment (park66665) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm]:
```suggestion
		if(floor_tile)
			new floor_tile(src)
```

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 5]
```diff
@@ -185,8 +185,16 @@
...
 /turf/open/floor/handle_decompression_floor_rip(sum)
-	if(sum > 20 && prob(CLAMP(sum / 10, 0, 30)))
-		remove_tile()
+	if(sum > 20 && prob(CLAMP(sum / 20, 0, 15)))
+		if(floor_tile)
+			new floor_tile(src)
+		make_plating()
+
+/turf/open/floor/plating/handle_decompression_floor_rip()
+	return
```
Comment (githubuser343) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:194]: What is the point of this empty return?

**Comment by ghost** [comment]
Might as well ask here, what does the "voting" and "vote passed" tag mean? Where does this voting take place and who votes?

**Comment by ike709** [comment]
image

**Review by ike709 — APPROVED** [review]

**Comment by PerishedFraud** [comment]
JUST REMOVE IT

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**